Added platform support for Nucleo-F334R8

matthiasbock opened the pull request:
Hello,

I implemented platform support for ST's Nucleo board
with the STM32F334R8 microcontroller:
http://www.st.com/en/evaluation-tools/nucleo-f334r8.html
Thought, you might be interested in it.

It compiles and I have confirmed the bootup
to booloader_main to be working.
Also CAN transmission seems to work.
Unfortunately I don't have the client working yet,
so I can't verify the overall process to be operational (yet).

Let me know, what you think and if you have change wishes.

Cheers, Matthias

matthiasbock committed:
Added platform support for Nucleo board with STM32F334R8 microcontroller

matthiasbock committed:
Added libopencm3 as build dependency in Makefile

matthiasbock committed:
Implemented Makefile parameter NO_COLOR

matthiasbock committed:
Moved pin/port definitions to platform.h
and LED handling to dedicated functions

matthiasbock committed:
Setting USE_CAN_ENABLE to disabled by default

antoinealb commented on platform/nucleo-board-stm32f334r8/platform.h:
```diff
@@ -0,0 +1,75 @@
+extern int application_address, application_size, config_page1, config_page2;
+
+
+// TODO: Superfluous(?) inline functions
```
They are not superflous, they are used by the bootloader to find the correct sections ;)

matthiasbock replied:
Any particular reason you are doing this with functions?
Couldn't you also be using the variables directly with "extern" ?

e.g. in commands.c:
````C
extern void* application_address;
````

antoinealb replied:
Yes, it could be done that way. Probably some legacy code, I can't remember any particular reason it was done that way. It was introduced in 7a2fb0f5 by @nuft maybe he remembers why he did it that way ?

antoinealb commented on platform/nucleo-board-stm32f334r8/include.mk:
```diff
@@ -0,0 +1,33 @@
+
```
We don't commit those files, as they are generated from the .yml files.

antoinealb reviewed (changes requested):
Hello,

Thanks for this platform!

Aside from the minor things I noted inline in the code, could you add the platform to the CI build (in `.travis.yml`), so that I can make sure it builds?

Also you say you don't have the client working yet, what is blocking you so far?

matthiasbock committed:
Implemented option to use inverted CAN_ENABLE signal

matthiasbock committed:
Deleted accidentally committed include.mk:
Should be generated using packager (https://github.com/cvra/packager).

Added include.mk to .gitignore

matthiasbock committed:
Added Nucleo-F334R8 platform to Travis CI build

matthiasbock committed:
Fixed STM32F334 linker section addresses and attributes

matthiasbock committed:
Added flash boundary variables to STM32F334 linker script

matthiasbock committed:
Fixed VFP register argument error
